tests/Feature/BookingTest.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Tests\Feature;
+
+use App\Models\Booking;
+use App\Models\Room;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Illuminate\Testing\Fluent\AssertableJson;
+use Tests\TestCase;
+
+class BookingTest extends TestCase
+{
+    use DatabaseMigrations;
+
+    public function test_bookRoom_success(): void
+    {
+        Booking::factory()->create();
+        Room::factory()->create();
+
+        $testData = [
+            "room_id" => 1,
+            "customer" => "Test customer",
+            "guests" => 2,
+            "start" => "2024-12-15",
+            "end" => "2024-12-25"
+        ];
+
+        $response = $this->post('/api/bookings', $testData);
+
+        $response->assertStatus(201)
+        ->assertJson(function (AssertableJson $json){
+            $json->hasAll(['message', 'data'])
+            ->has('data', function (AssertableJson $data) {
+                $data->hasAll([
+                    'room_id',
+                    'customer',
+                    'start',
+                    'end',
+                    'updated_at',
+                    'created_at',
+                    'id'
+                ])->whereAllType([
+                    'room_id' => 'integer',
+                    'customer' => 'string',
+                    'start' => 'string',
+                    'end' => 'string',
+                    'updated_at' => 'string|null',
+                    'created_at' => 'string|null',
+                    'id' => 'integer'
+                ]);
+            });
```

Comment:
As per my comment in the controller, this data shouldn't be in this response.

Reply:
What you should do however, is a database assertion to make sure the new booking was actually saved to the test DB